src/Kubernetes.WebSocket.cs

```diff
@@ -0,0 +1,195 @@
+            bool _shouldTrace = ServiceClientTracing.IsEnabled;
+            string _invocationId = null;
+            if (_shouldTrace)
+            {
+                _invocationId = ServiceClientTracing.NextInvocationId.ToString();
+                Dictionary<string, object> tracingParameters = new Dictionary<string, object>();
+                tracingParameters.Add("name", name);
+                tracingParameters.Add("@namespace", @namespace);
+                tracingParameters.Add("ports", ports);
+                tracingParameters.Add("cancellationToken", cancellationToken);
+                ServiceClientTracing.Enter(_invocationId, this, nameof(WebSocketNamespacedPodPortForwardAsync), tracingParameters);
+            }
+
+            // Construct URL
+            var uriBuilder = new UriBuilder(this.BaseUri);
+            uriBuilder.Scheme = this.BaseUri.Scheme == "https" ? "wss" : "ws";
+
+            if (!uriBuilder.Path.EndsWith("/"))
+            {
+                uriBuilder.Path += "/";
+            }
+
+            uriBuilder.Path += $"api/v1/namespaces/{@namespace}/pods/{name}/portforward";
+
+            foreach (var port in ports)
+            {
+                uriBuilder.Query += $"ports={port}&";
+            }
+
+            return StreamConnectAsync(uriBuilder.Uri, _invocationId, customHeaders, cancellationToken);
+        }
+
+        protected async Task<WebSocket> StreamConnectAsync(Uri uri, string invocationId = null, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            bool _shouldTrace = ServiceClientTracing.IsEnabled;
+
+            // Create WebSocket transport objects
+            WebSocketBuilder webSocketBuilder = this.CreateWebSocketBuilder();
+
+            // Set Headers
+            if (customHeaders != null)
+            {
+                foreach (var _header in customHeaders)
+                {
+                    webSocketBuilder.SetRequestHeader(_header.Key, string.Join(" ", _header.Value));
+                }
+            }
+
+            // Set Credentials
+            foreach (var cert in this.HttpClientHandler.ClientCertificates)
+            {
+                webSocketBuilder.AddClientCertificate(cert);
+            }
+
+            HttpRequestMessage message = new HttpRequestMessage();
+            await this.Credentials.ProcessHttpRequestAsync(message, cancellationToken);
+
+            foreach (var _header in message.Headers)
+            {
+                webSocketBuilder.SetRequestHeader(_header.Key, string.Join(" ", _header.Value));
+            }
+
+            // Send Request
+            cancellationToken.ThrowIfCancellationRequested();
+
+            WebSocket webSocket = null;
+
+            try
+            {
+                webSocket = await webSocketBuilder.BuildAndConnectAsync(uri, CancellationToken.None).ConfigureAwait(false);
+            }
+            catch (Exception ex)
```

Comment:
Please don't catch bare `Exception` only catch expected Exceptions.

Reply:
The catch catch statement is there so we can trace errors; the exception itself is rethrown.

.NET doesn't have the concept of checked exceptions and the MSDN documentation doesn't list which exceptions `ClientWebSocket.ConnectAsync` can throw.